workflow/libero_flow_poc/README.md

```diff
@@ -0,0 +1,252 @@
+
+`independent` : do you want this `Activity` to wait for previous list in this list to have completed before executing?
+
+`required` : if this `Activity` fails permanently should it cause the workflow to fail?
+
+`config` : an object containing key value configuration pairs. Please see the available config values for details.
+ 
+##### Activities
+An `Activity` is defined by subclassing the `BaseActivity` class in the `activities` package. The only thing you must implement is it's `do_activity` method. This is what will be executed by the worker processes. Below are examples of how you could implement the two activities defined in the above workflow definition: 
+
+`do_something_activity.py`
+```python
+from libero_flow.activities.base_activity import BaseActivity
+
+
+class DoSomethingActivity(BaseActivity):
+
+    def do_activity(self):
+        """Actually do some work"""
+        
+        num = 10
+        
+        if num > 20:
+            return self.PERMANENT_FAILURE
+        
+        return self.SUCCEEDED
+
+```
+
+`do_another_thing_activity.py`
+```python
+import requests
+
+from libero_flow.activities.base_activity import BaseActivity
+
+
+class DoAnotherThingActivity(BaseActivity):
+
+    def do_activity(self):
+        """Actually do some work"""
+        
+        response = requests.get('http://foobar.com/foo')
+        
+        if response.status_code == 200:
+            return self.SUCCEEDED
+        
+        return self.PERMANENT_FAILURE
+
+```
+
+You can place and retrieve values from the session store from within your activities using the `session_get` and `session_set` methods. 
+This is a way of passing data between activities.
+
+
+## Installation
+
+`docker-compose up --build`
+
+This will spin up the following containers:
+
+- Workflow API (Django application)
+- Web UI
+- Scheduler process
+- Decider process
+- Worker process
+- Postgres instance
+- Redis instance
+- RabbitMQ instance
+
+## Usage
+
+#### Starting a Workflow
+...
+
+Start a Workflow by message (via RabbitMQ using pika):
+```python
+import pika
+
+CREDENTIALS = pika.PlainCredentials('user', 'password')  # NOT REAL VALUES, USE CORRECT CREDENTIALS
+PARAMS = pika.ConnectionParameters(host='localhost', credentials=CREDENTIALS)
+
+connection = pika.BlockingConnection(parameters=PARAMS)
+
+
+with connection.channel() as channel:
+    msg = '{"name": "DummyWorkflow", "input_data": {}}'
+
+    channel.basic_publish(exchange='start_workflow',
+                          routing_key='workflow_starter',
+                          body=msg,
+                          properties=pika.BasicProperties(delivery_mode=2))
+
+```
+
+Start a Workflow by http request:
+```python
+
+import requests
+
+api_root = 'http://localhost:8000'  # this will depend on your workflow API server deployment 
+
+payload = {"name": "DummyWorkflow", "input_data": {}}
+url = api_root + '/workflows/api/v1/startworkflow/'
```

Comment:
this seems to be `start-workflow`:
```
curl -v http://localhost:8000/workflows/api/v1/start-workflow/ -d 'name=DummyWorkflow&input_data[key]=value'
{"message":"Start Workflow request sent"}
```
However after doing that, nothing is happening for me? No activity in http://localhost:4200/workflows or in Docker logs.

Reply:
I suspect you do not have any workflows defined. `DummyWorkflow` does not exist by default, it was just an example in the README.

You can create the `DummyWorkflow` as shown in the README, or just create a new one.